### PR TITLE
docs: Rewrite and expand Fleet Manager API endpoints documentation

### PR DESCRIPTION
**Pull request by miguelsorianod**
## Description

The current upstream documentation related to KFM Admin Endpoint was very sparse, which did not allow any user of KFM to understand how the KFM Admin endpoints work.
Aside from that, it contained details specific to how we configure it in Red Hat. The upstream repository of KFM and therefore its documentation should not contain any particularities of the KFM deployment(s) in Red Hat.

This PR rewrites and expands the Fleet Manager API endpoints documentation.
This will help users understand how the KFM admin endpoints work, how to configure them and how to interact with them.

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

**Comment by miguelsorianod** [comment]
I created a jira issue to track work on creating documentation for how KFM Admin API endpoints are configured and how to interact with them for our Red Hat specific deployments in stage and production. The issue is: https://issues.redhat.com/browse/MGDSTRM-11077

File: docs/admin-api-overview.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+Manager. The reason for that is that these endpoints allow
+performing potentially destructive actions, actions that should only be
+performed by Fleet Manager administrators, accessing sensitive information
+and accessing cross-tenant information. This set of endpoints are also often
+referred as Admin API endpoints.
+
+Authentication and Authorization is performed for the requests made against
+Admin API endpoints. OIDC is the protocol used to perform Authentication
+and Authorization. Therefore, when making a request to the Admin API endpoints
+an OIDC token has to be provided along it.
```
Comment (pawelpaszki) [inline docs/admin-api-overview.md:14]:
```suggestion
, an OIDC token has to be provided with such request.
```

File: docs/admin-api-overview.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+* `--admin-api-sso-endpoint-uri`: The URL path of the OIDC realm in the
+  authorization server specified in `--admin-api-sso-base-url`. By default
+  set to `/auth/realms/EmployeeIDP`
+* `--admin-api-sso-realm`: The name of the OIDC realm to be used in the
+  OIDC authorization server endpoint specified in `admin-api-sso-base-url`. By
+  default set to `EmployeeIDP`. The value specified here has to match the realm
+  name specified as part of the URL path specified in
+  `--admin-api-sso-endpoint-uri`
+
+Specifically, Fleet Manager Manager performs the following as part of the
```
Comment (pawelpaszki) [inline docs/admin-api-overview.md:37]:
```suggestion
Specifically, Fleet Manager performs the following as part of the
```

File: config/admin-authz-configuration.yaml [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
-# (https://rover.redhat.com/groups/) in order to grant access to the HTTP method containing that role
-# Configuration presented below is only used for testing purposes. The actual configuration deployed in 
-# production and stage environments will be provided in the saas template in app-interface
+# Each HTTP method allows configuring an arbitrary amount of roles
+# that authorize requests to the API. Any arbitrary role names can be configured.
+# When a request is performed to a Fleet Manager admin API endpoint Fleet
+# Manager checks if any of the role names specified for the used
+# HTTP method matches with any element in the `.realm_access.roles`
+# array, part of the JWT claims in the JWT token received.
+# If an HTTP method is not specified, or no roles are specified for a given
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply config/admin-authz-configuration.yaml:9]: Yes. I will add it :+1:

File: docs/admin-api-overview.md (hunk 2 above)
Comment (miguelsorianod) [reply docs/admin-api-overview.md]: updated

File: docs/admin-api-overview.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+and Authorization. Therefore, when making a request to the Admin API endpoints
+an OIDC token has to be provided along it.
+
+The available Fleet Manager API endpoints are documented in:
+* [KAS Fleet Manager Admin API OpenAPI specification file](../openapi/kas-fleet-manager-private-admin.yaml)
+* [Connector Fleet Manager Admin API OpenAPI specifications file](../openapi/connector_mgmt-private-admin.yaml)
+
+## Authentication
+
+To be able to perform Authentication in the Admin API endpoints the location of
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply docs/admin-api-overview.md]: updated

File: config/admin-authz-configuration.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply config/admin-authz-configuration.yaml:9]: added

File: config/admin-authz-configuration.yaml [hunk 5]
```diff
@@ -1,10 +1,16 @@
 ---
 # This file contains the role mapping for the admin API based on the HTTP methods.
-# Each HTTP method allows configuring an arbitrary amount of roles that authorize requests to the API.
-# Each of the role mapping must correspond to an existing Rover group 
-# (https://rover.redhat.com/groups/) in order to grant access to the HTTP method containing that role
-# Configuration presented below is only used for testing purposes. The actual configuration deployed in 
-# production and stage environments will be provided in the saas template in app-interface
+# Each HTTP method allows configuring an arbitrary amount of roles
+# that authorize requests to the API. Any arbitrary role names can be configured.
+# When a request is performed to a Fleet Manager admin API endpoint Fleet
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply config/admin-authz-configuration.yaml]: updated

**Review by pawelpaszki — APPROVED** [review]
looks good. thanks for adding it. feel free to wait for others to review or merge it

**Review by akoserwal — APPROVED** [review]

**Pull request merged**